### PR TITLE
#23: Clean up old connect jars automatically

### PR DESCRIPTION
**Pull request by bbaugher**
Test evidence, https://gist.github.cerner.com/BB021949/ef53782ad09ba480532ad31a48778bee

### CODE REVIEW

**Comment by noslowerdna** [comment]
+1

File: recipes/kafka-connect.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
 # frozen_string_literal: true
+
 include_recipe 'confluent::_install'
 connect_class = node['confluent']['kafka-connect']['distributed_mode'] ? node['confluent']['kafka-connect']['distributed_class'] : node['confluent']['kafka-connect']['standalone_class']
 
 confluent_extracted_dir = File.join(node['confluent']['install_dir'], "confluent-#{node['confluent']['version']}")
-connect_jdbc_jar_dir = File.join confluent_extracted_dir, 'share/java/kafka-connect-all'
+connect_share_dir = File.join confluent_extracted_dir, 'share', 'java/kafka-connect-all'
```
Comment (bjtieman) [inline recipes/kafka-connect.rb]: Curious: why break 'share' out but leave 'java' attached?

Comment (bbaugher) [reply recipes/kafka-connect.rb]: somehow didn't notice that, https://github.com/bbaugher/confluent/pull/24/commits/d99877ac6e073ef962171f4767388c8aa081720f

File: recipes/kafka-connect.rb [hunk 2]
```diff
@@ -37,6 +38,21 @@
   end
 end
 
+ruby_block 'Clean unreferenced Kafka Connect jars' do
+  block do
+    configured_jar_files = node['confluent']['kafka-connect']['jar_urls'].map { |url| File.basename(url) }
+    all_jar_files = Dir.entries(connect_share_dir)
+                       .select { |file| file != '.' && file != '..' }
```
Comment (bjtieman) [inline recipes/kafka-connect.rb]: reject { |file| file == '.' || file == '..' } might read easier...

File: recipes/kafka-rest.rb [hunk 3]
```diff
@@ -43,7 +44,7 @@
 end
 
 service 'kafka-rest' do
-  action [:enable, :start]
+  action %i[enable start]
```
Comment (bjtieman) [inline recipes/kafka-rest.rb:47]: Have you ever seen a good reason why this notation tends to be preferred?  I prefer the explicit array myself but have seen style guides say to prefer the % notation without really explaining why :(

Comment (bbaugher) [reply recipes/kafka-rest.rb:47]: This was a rubocop autocorrect. I have never seen that prior to this

File: spec/default_spec.rb [hunk 4]
```diff
@@ -48,6 +49,7 @@
     it 'should config JAAS' do
       chef_run.converge(described_recipe)
       expect(chef_run).to create_template('/opt/confluent/confluent-2.0.1/jaas.conf')
+      # rubocop:disable Lint/AmbiguousBlockAssociation
```
Comment (bjtieman) [inline spec/default_spec.rb:52]: The preferred style for multi-line blocks is do...end.  Does changing from {} to do...end remove the need for this?

**Comment by bjtieman** [comment]
A few nits and some for my information questions...+ 1 otherwise

File: recipes/kafka-connect.rb (hunk 2 above)
Comment (bbaugher) [reply recipes/kafka-connect.rb]: Yes thats better, https://github.com/bbaugher/confluent/pull/24/commits/9bf1a0499b5ef82d5a0391d76c79c6f242e4740d

File: spec/default_spec.rb (hunk 4 above)
Comment (bbaugher) [reply spec/default_spec.rb:52]: Did not realize thats what this was complaining about, fixed https://github.com/bbaugher/confluent/pull/24/commits/32cf7f244846c5c0eb98b20e7d2ca98dcfd7910e

Comment (bbaugher) [reply spec/default_spec.rb:52]: Actually had to revert this change as then rspec failed with `Must specify expected content or a block!` https://travis-ci.org/bbaugher/confluent/builds/236086089

**Comment by bbaugher** [comment]
Have enough +1s, merging

**Pull request merged**